examples/plot_custom_distributions.py

```diff
@@ -0,0 +1,76 @@
+"""
+======================
+Rolling out custom GLM
+======================
+
+This is an example demonstrating rolling out your custom
+GLM class using Pyglmnet.
+"""
+########################################################
+
+# Author: Pavan Ramkumar <pavan.ramkumar@gmail.com>
+# License: MIT
+
+from sklearn.model_selection import train_test_split
+from pyglmnet import GLMCV, datasets
+
+########################################################
+# Download and preprocess data files
+
+X, y = datasets.fetch_community_crime_data()
+n_samples, n_features = X.shape
+
+########################################################
+# Split the data into training and test sets
+
+X_train, X_test, y_train, y_test = \
+    train_test_split(X, y, test_size=0.33, random_state=0)
+
+########################################################
+# Now we define our own distribution class. This must
+# inherit from BaseDistribution. The BaseDistribution
+# class requires defining the following methods:
+#   - mu: inverse link function
+#   - grad_mu: gradient of the inverse link function
+#   - log_likelihood: the log likelihood function
+#   - grad_log_likelihood: the gradient of the log 
+#     likelihood.
+# All distributions in pyglmnet inherit from BaseDistribution
+#
+# This is really powerful. For instance, we can start from
+# the existing Binomial distribution and override mu and grad_mu
+# if we want to use the cloglog link function.
+
+import numpy as np
+from pyglmnet.distributions import Binomial
+
+
+class CustomBinomial(Binomial):
```

Comment:
@pavanramkumar I'm not sure this is legit. The `log_likelihood` function for `Binomial` does not call `self.mu` or does it? Is the log likelihood the log likelihood taking into account the link function?